packages/story-editor/src/app/story/useStoryReducer/reducers/combineElements.js

```diff
@@ -53,115 +54,87 @@ import { removeAnimationsWithElementIds } from './utils';
 
-  // Elements are now
-  const elements = page.elements
-    // Remove first element if combining from existing id
-    .filter(({ id }) => id !== firstId)
-    // Update reference to second element
-    .map((el) => (el.id === secondId ? newElement : el));
-
-  // First element should always be the image getting applied to
-  // new element. We want to remove any animations it has. Second
-  // element should be an element with or without animations that
-  // we want to retain.
-  //
-  // We want different behavior for copy and paste where we
-  // replace the element's animation with any coming from the
-  // newly pasted element.
-  const newAnimations = removeAnimationsWithElementIds(
-    page.animations,
-    shouldRetainAnimations ? [firstId] : [firstId, secondId]
-  );
+    if (!element || !element.resource || !secondElement) {
+      return;
+    }
 
-  const newPage = {
-    ...page,
-    elements,
-    ...newPageProps,
-    animations: newAnimations,
-  };
+    if (secondElement.isDefaultBackground) {
+      page.defaultBackgroundElement = {
+        ...secondElement,
+        // But generate a new ID for this temp background element
+        id: uuidv4(),
+      };
+    }
 
-  const newPages = [
-    ...state.pages.slice(0, pageIndex),
-    newPage,
-    ...state.pages.slice(pageIndex + 1),
-  ];
+    const propsFromFirst = [
+      'alt',
+      'type',
+      'resource',
+      'scale',
+      'focalX',
+      'focalY',
+      'tracks',
+      'poster',
+    ];
+
+    // If the element we're dropping into is not background, maintain link and border.
+    if (!secondElement.isBackground) {
+      propsFromFirst.push('link');
+      propsFromFirst.push('border');
+      if (canSupportMultiBorder(secondElement)) {
+        propsFromFirst.push('borderRadius');
+      }
+    } else {
+      // If we're dropping into background, maintain the flip and overlay, too.
+      // Only copy position properties for backgrounds, as they're ignored while being background
+      // For non-backgrounds, elements should keep original positions from secondElement
+      propsFromFirst.push('flip', 'overlay', 'width', 'height', 'x', 'y');
+    }
 
-  return {
-    ...state,
-    pages: newPages,
-    selection: [secondId],
-  };
-}
+    const newElement = {
+      // First copy everything from existing element except if it was default background
+      ...objectWithout(secondElement, ['isDefaultBackground']),
+      // Then set sensible default attributes
+      ...DEFAULT_ATTRIBUTES_FOR_MEDIA,
+      // Then copy all relevant attributes from new element
+      ...objectPick(element, propsFromFirst),
+    };
+
+    // Elements are now
+    page.elements = page.elements
+      // Remove first element if combining from existing id
+      .filter(({ id }) => id !== firstId)
+      // Update reference to second element
+      .map((el) => (el.id === secondId ? newElement : el));
+
+    // First element should always be the image getting applied to
```

Comment:
These comments are really helpful - to give some back story into what is happening.

💯